### PR TITLE
Implemented Issue #39. Drag and Drop feature for commit process

### PR DESCRIPTION
**Pull request by aujlarj**
Implemented Drag and Drop feature by adding by using HTML5 drag and drop functionality.
[Issue-39](https://github.com/uvicgse/project-virtual-team-2/issues/39)
[Documentation](https://github.com/uvicgse/project-virtual-team-2/wiki/Drag-and-Drop-Feature-Documentation)

Now Staged and UnStaged files can be dragged into each other's section.

### CODE REVIEW

**Review by jbenning97 — APPROVED** [review]
**Does the code work? Did I test it locally? What OS am I using? Does it perform its intended function, the logic is correct etc.**

Yes, tested on macOS High Sierra 10.13.3, drag and drop is working. Didn't break committing or other staging functionality. Handles dropping outside of zones gracefully. Great implementation.

**Is the code well documented? Would it be easily understood to other developers?**

yes, appropriate documentation/comments

**Does it have correct linting/formatting?**

Yes

**Is there any redundant or duplicate code? Any hard-coded values?**

No

**Can any of the code be replaced with library functions?**

No

**Are good names used for methods, variables, etc.? Do they follow naming conventions?**

Yes

**Were any other issues found?**

Not that I could find

**Comment by aujlarj** [comment]
@jbenning97 I will soon update CSS, to look the dragging element more noticeable.

File: app/misc/git.ts [hunk 1]
```diff
@@ -1742,6 +1742,41 @@ function displayModifiedFiles() {
...
+            const stagepanel = document.querySelector("div.staged-files-header");
+            // create top and bottom boundary for drag release
+            const topboundary = stagepanel.getBoundingClientRect();
+            const bottomboundary = filepanel.getBoundingClientRect();
+
+            // mouse pointer location while dragging
+            var x_axis = e.clientX;
+            var y_axis = e.clientY;
+
+            //console.log(topboundary);
```
Comment (Leon0315) [inline app/misc/git.ts:1763]: delete if not using

**Review by Leon0315 — APPROVED** [review]
**Does the code work? Did I test it locally? What OS am I using? Does it perform its intended function, the logic is correct etc.**
Yes, tested on macOS 10.14. Drag and drop works for file, better if add a 🚫sign when drag outside of the panel. Commit and push works.

**Is the code well documented? Would it be easily understood to other developers?**

yes, delete unused lines.

**Does it have correct linting/formatting?**

Yes

**Is there any redundant or duplicate code? Any hard-coded values?**
There's a small duplicate function in both method, but it's alright.

**Can any of the code be replaced with library functions?**
Easier with HTML drag and drop

**Are good names used for methods, variables, etc.? Do they follow naming conventions?**

Yes

**Were any other issues found?**

No

**Pull request merged**